### DISCUSSION TITLE
How to create reusable attributes in block development?

### DISCUSSION THREAD

**Discussion by chetansharma10** [WordPress/gutenberg#54644] [General]
I have a project in which I have row and column block but i just want to use padding attribute in both blocks without writing again & again in both blocks.I know there is a support api and I used the padding attribute dynamically using filter registerBlockType.I want to know that whether this method is ok to go.Please tell

**Comment by tresorama** [comment]
It is not clear what your problem is about.
Please include pseudo code and a detailed explanation of what you cannot achieve so someone can help you if possible...

**Reply by chetansharma10** [reply]
### Example
Here I have two block.json of row and column block.And In these two files i write same attributes of **padding** repeatedly
### Block Row
`
{
	"apiVersion": 2,
	"name": "xyz/row",
	"title": "Row",
	"attributes": {
		"padding_top": {
			"type": "string",
			"default": null
		},
		"padding_right": {
			"type": "string",
			"default": null
		},
		"padding_bottom": {
			"type": "string",
			"default": null
		},
		"padding_left": {
			"type": "string",
			"default": null
		},
	},
	"supports": {
	},
}`

### Block Column
`{
	"apiVersion": 2,
	"name": "xyz/column",
	"title": "Column",
	"attributes": {
		"padding_top": {
			"type": "string",
			"default": null
		},
		"padding_right": {
			"type": "string",
			"default": null
		},
		"padding_bottom": {
			"type": "string",
			"default": null
		},
		"padding_left": {
			"type": "string",
			"default": null
		},
	},
}`

What I need ->
`
{
	"apiVersion": 2,
	"name": "xyz/row",
	"title": "Row",
	"attributes": {
	},
	"supports": {
		"padding":true
	},
}`

and in column block
`{
	"apiVersion": 2,
	"name": "xyz/column",
	"title": "Column,
	"attributes": {
	},
	"supports": {
		"padding":true
	},
}`

Is there is something that inject attributes dynamically like support does?
